src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalMessagesOpt.java

```diff
@@ -4,24 +4,47 @@
  */
 package com.vmware.vipclient.i18n.messages.api.opt.local;
 
+import java.util.Map;
+
 import org.json.simple.JSONObject;
 
+import com.vmware.vipclient.i18n.VIPCfg;
+import com.vmware.vipclient.i18n.base.cache.MessageCacheItem;
+import com.vmware.vipclient.i18n.messages.api.opt.MessageOpt;
 import com.vmware.vipclient.i18n.messages.api.opt.Opt;
 import com.vmware.vipclient.i18n.messages.dto.MessagesDTO;
+import com.vmware.vipclient.i18n.util.FormatUtils;
 import com.vmware.vipclient.i18n.util.JSONBundleUtil;
 
-public class LocalMessagesOpt implements Opt {
+public class LocalMessagesOpt implements Opt, MessageOpt {
+	
+	private static final String OFFLINE_RESOURCE_PATH = "{0}/messages_{1}.json";
     private MessagesDTO dto;
 
     public LocalMessagesOpt(MessagesDTO dto) {
         this.dto = dto;
     }
 
+    @Deprecated
     public JSONObject getComponentMessages() {
         return JSONBundleUtil.getMessages(dto.getLocale(), dto.getProductID(),
                 dto.getVersion(), dto.getComponent());
     }
+    
+    @Override
+    public void getComponentMessages(MessageCacheItem cacheItem) {
+    	String pathName = VIPCfg.getInstance().getOfflineResourcesBaseUrl();
+    	String filePath = FormatUtils.format(OFFLINE_RESOURCE_PATH, dto.getComponent(), dto.getLocale());
+    	if (pathName != null) {
+    		pathName = pathName.concat(filePath);
```

Comment:
I meant line 39. 
Wtich Paths.get the, the '/' in offlineResourcesBaseUrl of VIPCfg will have no affect on the result. 
Please check.

Reply:
> By new flow, client will fallback to local bundles if failing to get from server.
> I couldn't find changes on this. Could you check this?

Sure.  I was planning to check it in the next Pr so that you are not overwhelmed. I have just checked it in now.